From 8fe6eb6866ff84ef952095a57897fa9f024b33ae Mon Sep 17 00:00:00 2001
From: Kiran Rudramuni <kchitrik@codeaurora.org>
Date: Thu, 19 Feb 2015 11:12:58 -0800
Subject: [PATCH] init.qcom.uicc: Disable uicc property for the Liquid case

uicc_insert() and uicc_remove() actions are only
performed when sys.usb_uicc.loading property is
set to 1 by USB uicc client. This is not relevant
for Liquid case.

Change-Id: Ifbaa5b69917c674f93708fe6baaaf112296b9e2d
---
 rootdir/etc/init.qcom.uicc.sh | 11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

diff --git a/rootdir/etc/init.qcom.uicc.sh b/rootdir/etc/init.qcom.uicc.sh
index 5129ee7..604fdff 100644
--- a/rootdir/etc/init.qcom.uicc.sh
+++ b/rootdir/etc/init.qcom.uicc.sh
@@ -1,5 +1,5 @@
 #!/system/bin/sh
-# Copyright (c) 2014, The Linux Foundation. All rights reserved.
+# Copyright (c) 2014-2015, The Linux Foundation. All rights reserved.
 #
 # Redistribution and use in source and binary forms, with or without
 # modification, are permitted provided that the following conditions are met:
@@ -30,8 +30,15 @@ target=`getprop ro.board.platform`
 action=`getprop sys.usb_uicc.enabled`
 uicc_loading=`getprop sys.usb_uicc.loading`
 
+# Set platform variables
+if [ -f /sys/devices/soc0/hw_platform ]; then
+    soc_hwplatform=`cat /sys/devices/soc0/hw_platform` 2> /dev/null
+else
+    soc_hwplatform=`cat /sys/devices/system/soc/soc0/hw_platform` 2> /dev/null
+fi
+
 # Perfom uicc_insert/uicc_remove only when usb uicc client says so
-if [ $uicc_loading != "1" ]; then
+if [ ($soc_hwplatform != "MTP") -o ($uicc_loading != "1") ]; then
     action=""
 fi
 
-- 
1.8.2.1

